### PR TITLE
export drawer's open/close callback

### PR DESCRIPTION
**Pull request by msyaojz**

### CODE REVIEW

File: packages/flutter/lib/src/material/scaffold.dart [hunk 1]
```diff
@@ -842,6 +844,9 @@ class Scaffold extends StatefulWidget {
   /// The default value of this property, like the default value of
   /// [AppBar.primary], is true.
   final bool primary;
+  /// Optional callback that is called when a [Drawer] is opened or closed.
+  final DrawerCallback drawerCallback;
+  final DrawerCallback endDrawerCallback;
```
Comment (sir-boformer) [inline packages/flutter/lib/src/material/scaffold.dart]: This line also needs documentation, and there should be an empty line between every field declaration

**Comment by Jon-Salmon** [comment]
Any development on this @msyaojz 
The lack of this feature is currently blocking me from release, so I may need to open a seperate PR for the same isssue soon.

**Comment by Hixie** [comment]
This is probably not the approach we would want to take for this... I would imagine we would either make `openDrawer` return a `Future` that's also exposed on `ScaffoldState`, or maybe have a `ValueListenable<bool>` that reports the current state of each drawer, or some such. But it mostly depends on what the use cases are. I recommend going into more detail in one of the issues linked above about what exactly it is that you need this feature for so that we can come up with a more coherent design.

**Comment by Hixie** [comment]
Another possible idea is exposing the `Animation` that represents the drawers sliding in and out.

(The problem with callbacks is that they wouldn't be usable in other contexts, e.g. you couldn't await one, or pass one to an `AnimationBuilder`, or the like.)

**Comment by jslavitz** [comment]
Because this implementation isn't really the best way of implementing this feature, we're going to close this pull request for now.

**Comment by Hixie** [comment]
Since the CLA wasn't signed on this PR, I'm going to close it. We can't accept patches where the CLA hasn't been signed.

@msyaojz if you have the opportunity to sign the CLA, please don't hesitate to do so and open a new PR! Thanks!

**Comment by bikcrum** [comment]
This should be resolved by #67249

**Pull request closed without merging**